engine/src/witness/btc/deposits.rs

```diff
@@ -55,25 +55,30 @@ impl<Inner: ChunkedByVault> ChunkedByVaultBuilder<Inner> {
 		self.then(move |epoch, header| {
 			let process_call = process_call.clone();
 			async move {
+				// TODO: Make addresses a Map of some kind?
+				let ((((), txs), deposit_channels), private_channels) = header.data;
+
 				let vault_addresses = {
 					use cf_chains::btc::{
 						deposit_address::DepositAddress, AggKey, CHANGE_ADDRESS_SALT,
 					};
 
 					let key: &AggKey = &epoch.info.0;
 
-					let maybe_previous_vault_address =
-						key.previous.map(|key| DepositAddress::new(key, CHANGE_ADDRESS_SALT));
-					let current_vault_address =
-						DepositAddress::new(key.current, CHANGE_ADDRESS_SALT);
-
-					[current_vault_address].into_iter().chain(maybe_previous_vault_address)
+					[key.current].into_iter().chain(key.previous).flat_map(|key| {
```

Comment:
this block is reasonably clear if you know the context, but without it, might be a little hard to parse, so maybe a little comment reasoning it through here